samza-kafka/src/test/scala/org/apache/samza/system/kafka/TestKafkaSystemAdmin.scala

```diff
@@ -301,12 +260,6 @@ class TestKafkaSystemAdmin {
     val spec = StreamSpec.createCoordinatorStreamSpec(topic, "kafka")
     systemAdmin.createStream(spec)
     validateTopic(topic, 1)
-    val topicMetadataMap = TopicMetadataCache.getTopicMetadata(Set(topic), "kafka", metadataStore.getTopicInfo)
-    assertTrue(topicMetadataMap.contains(topic))
-    val topicMetadata = topicMetadataMap(topic)
-    val partitionMetadata = topicMetadata.partitionsMetadata.head
-    assertEquals(0, partitionMetadata.partitionId)
-    assertEquals(3, partitionMetadata.replicas.size)
```

Comment:
Yeah. I tried to keep this by using APIs from `adminClient` and `adminZkClient`but in vain. I'd have to replicate the logic in the file `ClientsUtilTopicMetadataStore.scala` that was removed as part of the PR. Since we no longer have any active code path within our codebase that reads this data, I chose to remove it.

Reply:
The end-to-end check seems unnecessary anyways for a unit test.
However, since you are removing that check, it might be good to have a test which uses Mockito to verify that the correct configurations are passed to the admin client when creating topics. If I recall correctly, in the past, we may have had some issues which involved replication factors (not sure if they were this exact flow though).

Reply:
Agreed. Let me do the Mockito way 👍